### PR TITLE
Add couple of tests for Tags

### PR DESCRIPTION
**Pull request by atodorov**
@clumens I managed to get the new tests working but I don't quite have an idea what I'm doing here.

`mkTag` seems to read a byte stream (the RPM contents I presume) and returns different kinds of data based on which chunks it reads (am I right?). However I have no idea what the actual data in the byte stream should look like so I've made up some values just to satisfy the assertions.

### CODE REVIEW

**Comment by clumens** [comment]
Check out where `mkTag` is used - in the `parseOneTag` function in `Parse.hs`.  I'm not sure how familiar you are with RPM, but basically all the tags that make up the metadata of the RPM are stored in an array known as the headers.  For each element of the array, you have a number that tells you what the tag is, a number that tells you its type (string, number, whatever), a number that tells you where in the header to start reading values for this tag, and a number telling you how many values there are.

lib/tagtbl.C in the RPM source is helpful for figuring out what's valid and what's not.  Running the inspect program out of haskell-rpm on some RPMs is also helpful for seeing how things are structured.

`mkTag` just uses a lot of little helper functions to read some bytes out of the header array, convert it to the right type, and make a Haskell value out of it.

File: tests/RPM/TagsSpec.hs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    it "returns FileModes for 1030 `tag' value" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 3 0 2
+      tag `shouldBe` Just (FileModes [0,5])
+
+    it "returns Nothing for 1030 `tag' value and `ty' != 3" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 0 0 2
+      tag `shouldBe` Nothing
```
Comment (clumens) [inline tests/RPM/TagsSpec.hs]: I think you are on the right track here.  Certainly we don't want tests for every single tag, but it'd be worth testing each type of tag and their corner cases (empty lists, count is 0, that kind of thing).

Comment (atodorov) [inline tests/RPM/TagsSpec.hs]: I can do a loop over different tag values which are handled in the same way and then define the hspec examples for `x` :).

I will checkout the RPM code to get an idea about meaningful values to use for testing. Otherwise I will be testing with zeroes most likely.

**Comment by atodorov** [comment]
@clumens how about the updated version? 

I hate it that I have to duplicate the Tag constructors in the expectation but that is the only was I was able to make Haskell actually evaluate these expressions (check out the coverage report). It may not be so bad after all.

**Comment by dashea** [comment]
Only issue I have is the hlint warnings causing the checks to fail. I don't care about the warnings themselves, so just annotating them out is fine.

**Comment by dashea** [comment]
> just annotating them out is fine

for what it's worth, setting the annotation on `spec` doesn't seem to work, but

`{-# ANN module "HLint: ignore Reduce duplication" #-}`

does.
